airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalAttemptExecution.java

```diff
@@ -144,6 +138,22 @@ public OUTPUT get() {
     }
   }
 
+  private void saveWorkflowIdForCancellation() throws IOException {
+    // If the jobId is not a number, it means the job is a synchronous job. No attempt is created for
+    // it, and it cannot be cancelled, so do not save the workflowId. See SynchronosSchedulerClient.java
+    // for info.
+    if (NumberUtils.isCreatable(jobRunConfig.getJobId())) {
```

Comment:
@jrhizor I think because these jobs aren't saved in the database at all there would be no place to save the workflow id.

Reply:
What I couldn't figure out is how to retrieve the workflow id outside of the temporal workflow. I don't think that is possible since we aren't assigning a custom label/workflow id without standing up elastisearch. So we need the workflow to first be created (Temporal assigns an id) and then save it.

Reply:
Spoke offline with Jared. This is the next phase of work: https://github.com/airbytehq/airbyte/issues/5936